cocotb/share/lib/fli/FliCbHdl.cpp

```diff
@@ -140,6 +144,53 @@ int FliStartupCbHdl::arm_callback()
     return 0;
 }
 
+
+static std::vector<std::string> get_argv()
+{
+    /*  Necessary to implement PLUSARGS
+        There is no function available on the FLI to obtain argc+argv directly from the
+        simulator. To work around this we use the TCL interpreter that ships with Questa,
+        some TCL commands, and the TCL variable `argv` to obtain the simulator argc+argv.
+    */
+    std::vector<std::string> argv;
+
+    // obtain a reference to TCL interpreter
+    Tcl_Interp* interp = reinterpret_cast<Tcl_Interp*>(mti_Interp());
+
+    // get argv TCL variable
+    if (mti_Cmd("return -level 0 $argv") != TCL_OK) {
+        const char* errmsg = Tcl_GetStringResult(interp);
+        LOG_WARN("Failed to get reference to argv: %s", errmsg);
+        Tcl_ResetResult(interp);
+        return argv;
+    }
+    Tcl_Obj* result = Tcl_GetObjResult(interp);
+    Tcl_IncrRefCount(result);
+    Tcl_ResetResult(interp);
+
+    // split TCL list into length and element array
+    int argc;
+    Tcl_Obj** tcl_argv;
+    if ((Tcl_ListObjGetElements(interp, result, &argc, &tcl_argv) != TCL_OK) || (argc == 0)) {
+        const char* errmsg = Tcl_GetStringResult(interp);
```

Comment:
What value will this hold if `argc == 0`?

Reply:
I think it would be best to split this `if` into two.

Reply:
I think we can remove the second check. `argc == 0` and the function returns `TCL_OK` only when the list is empty, `tcl_argv == NULL`. However, the way this function is written means we will never dereference that `NULL` pointer.